### ISSUE TITLE
Adding `ContentType` to S3Client's 'PutObjectCommand' gives 'SignatureDoesNotMatch' Error

### ISSUE THREAD

**Issue by bhavyasaggi** [aws/aws-sdk-js-v3#1916]
Confirm by changing [ ] to [x] below to ensure that it's a bug:
- [x] I've gone through [Developer Guide](https://docs.aws.amazon.com/sdk-for-javascript/v2/developer-guide/welcome.html) and [API reference](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/)
- [x] I've checked [AWS Forums](https://forums.aws.amazon.com) and [StackOverflow](https://stackoverflow.com/questions/tagged/aws-sdk-js) for answers
- [x] I've searched for [previous similar issues](https://github.com/aws/aws-sdk-js/issues) and didn't find any solution
  
**Describe the bug**
I'm Pulling files from Google Cloud Storage and passing them to S3Client PutObjectCommand to host on S3. However, the Content-Type is defaulted to `application/octet-stream`.
Since I need to host, I require the Content-Type to be 'text/html' or whatsoever (for which I use `mime-type` to get appropriate mime from file/extension)
Upon providing with ContentType, the code fails with 
`"SignatureDoesNotMatch: The request signature we calculated does not match the signature you provided. Check your key and signing method."`

Eg.
```
>  SignatureDoesNotMatch: The request signature we calculated does not match the signature you provided. Check your key and signing method.
>      at deserializeAws_restXmlPutObjectCommandError (C:\Users\Carbon\Desktop\codebase\boot-gear\functions\node_modules\@aws-sdk\client-s3\dist\cjs\protocols\Aws_restXml.js:8035:41)
>      at runMicrotasks (<anonymous>)
>      at processTicksAndRejections (internal/process/task_queues.js:97:5)
>      at async C:\Users\Carbon\Desktop\codebase\boot-gear\functions\node_modules\@aws-sdk\middleware-serde\dist\cjs\deserializerMiddleware.js:7:20
>      at async C:\Users\Carbon\Desktop\codebase\boot-gear\functions\node_modules\@aws-sdk\middleware-signing\dist\cjs\middleware.js:12:24
>      at async StandardRetryStrategy.retry (C:\Users\Carbon\Desktop\codebase\boot-gear\functions\node_modules\@aws-sdk\middleware-retry\dist\cjs\defaultStrategy.js:55:46)       
>      at async C:\Users\Carbon\Desktop\codebase\boot-gear\functions\node_modules\@aws-sdk\middleware-logger\dist\cjs\loggerMiddleware.js:7:22
>      at async Promise.all (index 2)
>      at async C:\Users\Carbon\Desktop\codebase\boot-gear\functions\helpers\aws.js:52:3 
>      at async C:\Users\Carbon\Desktop\codebase\boot-gear\functions\index.js:369:9 {    
>    Code: 'SignatureDoesNotMatch',
>    AWSAccessKeyId: '*******',
>    StringToSign: 'AWS4-HMAC-SHA256\n' +
>      '*********\n' +
>      '20210110/ap-south-1/s3/aws4_request\n' +
>      'c02d08456b15aed786025fd0a598eb950477695e45bbb6b39ce115d209e0929f',
>    SignatureProvided: 'c65b512356cd0e2ab378deb3848c26110eff92b1e610e90de056bd9f7ae98f66',
>    StringToSignBytes: '41 57 53 34 2d 48 4d 41 43 2d 53 48 41 32 35 36 0a 32 30 32 31 30 31 31 30 54 31 34 33 30 33 34 5a 0a 32 30 32 31 30 31 31 30 2f 61 70 2d 73 6f 75 74 68 
2d 31 2f 73 33 2f 61 77 73 34 5f 72 65 71 75 65 73 74 0a 63 30 32 64 30 38 34 35 36 62 31 35 61 65 64 37 38 36 30 32 35 66 64 30 61 35 39 38 65 62 39 35 30 34 37 37 36 39 35 65 34 35 62 62 62 36 62 33 39 63 65 31 31 35 64 32 30 39 65 30 39 32 39 66',
>    CanonicalRequest: 'PUT\n' +
>      '/randombakwas.com/bin/settlements-2.png\n' +
>      'x-id=PutObject\n' +
>      'amz-sdk-invocation-id:03cbd46d-4c41-4e2b-8a65-4c741a9023ce\n' +
>      'amz-sdk-request:attempt=3; max=3\n' +
>      'content-length:1288415\n' +
>      'content-type:image/png\n' +
>      'host:s3.ap-south-1.amazonaws.com\n' +
>      'x-amz-content-sha256:300e1bf606fb5b26ed5277a96055ee8a44a9fdb34969fb96df84511fd42e168d\n' +
>      'x-amz-date:20210110T143034Z\n' +
>      '\n' +
>      'amz-sdk-invocation-id;amz-sdk-request;content-length;content-type;host;x-amz-content-sha256;x-amz-date\n' +
>      '300e1bf606fb5b26ed5277a96055ee8a44a9fdb34969fb96df84511fd42e168d',
>    CanonicalRequestBytes: '50 55 54 0a 2f 72 61 6e 64 6f 6d 62 61 6b 77 61 73 2e 63 6f 
6d 2f 62 69 6e 2f 73 65 74 74 6c 65 6d 65 6e 74 73 2d 32 2e 70 6e 67 0a 78 2d 69 64 3d 50 75 74 4f 62 6a 65 63 74 0a 61 6d 7a 2d 73 64 6b 2d 69 6e 76 6f 63 61 74 69 6f 6e 2d 69 64 3a 30 33 63 62 64 34 36 64 2d 34 63 34 31 2d 34 65 32 62 2d 38 61 36 35 2d 34 63 37 34 
31 61 39 30 32 33 63 65 0a 61 6d 7a 2d 73 64 6b 2d 72 65 71 75 65 73 74 3a 61 74 74 65 6d 70 74 3d 33 3b 20 6d 61 78 3d 33 0a 63 6f 6e 74 65 6e 74 2d 6c 65 6e 67 74 68 3a 31 32 38 38 34 31 35 0a 63 6f 6e 74 65 6e 74 2d 74 79 70 65 3a 69 6d 61 67 65 2f 70 6e 67 0a 68 
6f 73 74 3a 73 33 2e 61 70 2d 73 6f 75 74 68 2d 31 2e 61 6d 61 7a 6f 6e 61 77 73 2e 63 6f 6d 0a 78 2d 61 6d 7a 2d 63 6f 6e 74 65 6e 74 2d 73 68 61 32 35 36 3a 33 30 30 65 31 62 66 36 30 36 66 62 35 62 32 36 65 64 35 32 37 37 61 39 36 30 35 35 65 65 38 61 34 34 61 39 
66 64 62 33 34 39 36 39 66 62 39 36 64 66 38 34 35 31 31 66 64 34 32 65 31 36 38 64 0a 78 2d 61 6d 7a 2d 64 61 74 65 3a 32 30 32 31 30 31 31 30 54 31 34 33 30 33 34 5a 0a 0a 61 6d 7a 2d 73 64 6b 2d 69 6e 76 6f 63 61 74 69 6f 6e 2d 69 64 3b 61 6d 7a 2d 73 64 6b 2d 72 
65 71 75 65 73 74 3b 63 6f 6e 74 65 6e 74 2d 6c 65 6e 67 74 68 3b 63 6f 6e 74 65 6e 74 2d 74 79 70 65 3b 68 6f 73 74 3b 78 2d 61 6d 7a 2d 63 6f 6e 74 65 6e 74 2d 73 68 61 32 35 36 3b 78 2d 61 6d 7a 2d 64 61 74 65 0a 33 30 30 65 31 62 66 36 30 36 66 62 35 62 32 36 65 
64 35 32 37 37 61 39 36 30 35 35 65 65 38 61 34 34 61 39 66 64 62 33 34 39 36 39 66 62 39 36 64 66 38 34 35 31 31 66 64 34 32 65 31 36 38 64',
>    RequestId: 'BFFFB3186F28ED87',
>    HostId: 'iLF865akMuO98onn4mGIZBZ/wptUH3cOIy8qy4OnBY/m35w5NS8Oh0qDYFPl5+yqiT/DHiX5LzM=',
>    '$fault': 'client',
>    '$metadata': {
>      httpStatusCode: 403,
>      httpHeaders: {
>        'x-amz-request-id': 'BFFFB3186F28ED87',
>        'x-amz-id-2': 'iLF865akMuO98onn4mGIZBZ/wptUH3cOIy8qy4OnBY/m35w5NS8Oh0qDYFPl5+yqiT/DHiX5LzM=',
>        'content-type': 'application/xml',
>        'transfer-encoding': 'chunked',
>        date: 'Sun, 10 Jan 2021 14:30:34 GMT',
>        connection: 'close',
>        server: 'AmazonS3'
>      },
>      requestId: undefined,
>      attempts: 3,
>      totalRetryDelay: 135
>    }
>  }
```

**Is the issue in the browser/Node.js?**
Node.js

**Details of the browser/Node.js version**
Node v12.18.2

**SDK version number**
"@aws-sdk/client-s3": "^3.0.0",

**To Reproduce (observed behavior)**
```
// 
const storageFilesMap = {}
await Promise.all(
    storageItems.map((storageItem) =>
      storageItem
        .download()
        .then((downloadResponse) => downloadResponse[0])
        .then(async (downloadBlob) => {
          const [{ name }] = await storageItem.getMetadata()
          const storageKey = name.replace(
            new RegExp(`^${storagePrefixKey}`),
            ''
          )
          storageFilesMap[storageKey] = downloadBlob
          return true
        })
    )
  )
//
await Promise.all(
    Object.keys(storageFilesMap).map((storageKey) =>
      s3.send(
        new PutObjectCommand({
          Bucket: bucketName,
          Key: storageKey,
          Body: storageFilesMap[storageKey],
          ContentType: mime.lookup(storageKey),
        })
      )
    )
  )
```

**Expected behavior**
File is to be uploaded with set Content-Type when provided.

**Comment by ellemenno** [comment]
@bhavyasaggi was this fixed for you in [v3.3.0](https://github.com/aws/aws-sdk-js-v3/releases/tag/v3.3.0) (seems similar to #1800) ?

**Comment by rati-dzidziguri** [comment]
Same issue when providing contentType inside PutObjectCommand and this is a critical bug as we are not able to dynamically generate objects to put inside the bucket any update on this?

**Comment by bhavyasaggi** [comment]
Verified. Closing as this is fixed in v3.3.0.
Thanks to all for the assist!!
